lib/libzpool/uzfs_io.c

```diff
@@ -218,3 +253,16 @@ uzfs_flush_data(zvol_state_t *zv)
 {
 	zil_commit(zv->zv_zilog, ZVOL_OBJ);
 }
+
+void
+uzfs_set_rebuilding_mode(zvol_state_t *zv)
+{
+	zv->in_rebuilding_mode = B_TRUE;
+	zv->rebuild_bytes = 0;
+}
+
+void
+uzfs_unset_rebuilding_mode(zvol_state_t *zv)
+{
+	zv->in_rebuilding_mode = B_FALSE;
+}
```

Comment:
synchronization between setting rebuild mode and IO path should be handled by caller API only. Since IO path can be parallel so i have avoided any synchronization at this level.